Added recommended settings for multiple games (see comment).

MarioManTAW opened the pull request:
Animal Crossing: City Folk, Donkey Kong Country Returns, Kirby's Epic Yarn, Mario Kart Wii, Mario Sports Mix, Metroid Prime 3: Corruption, New Super Mario Bros. Wii, Punch-Out, Super Paper Mario, Super Smash Bros. Brawl, Wii Music, Wii Sports, and Wii Sports Resort

I think I've now covered most of the 1st-party, made-for-Wii games.
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/dolphin-emu/dolphin/3735)
<!-- Reviewable:end -->

MaJoRoesch commented:
Can you please specify the game in the title? The title is a bit confusing. :/

JMC47 commented:
Please add comments to the INI saying what the changes fix/affect, so that people in the future can easily test if options are changed or removed.

MaJoRoesch commented:
@Linktothepast Please review!

JMC47 commented:
Not trying to be rude or grill you, but, please elaborate on setting OpenGL to true.  D3D and OpenGL should be identical in compatibility, and if they aren't, I request you make an issue and link to that in the INI so that we have a frame of reference as to why it's unstable and more details.

One of the biggest problems I have with the INIs as they are now is that there are a ton of decisions from years ago that I can't verify as to why, and saying OpenGL for stability without saying how/why/what happens differently is still not enough for me to verify quickly if an issue is fixed.

MaJoRoesch commented:
It also varies depending on the specific hardware.

MarioManTAW commented:
What game was that on again? I'm just basing my changes on the wiki, so a large number of these are issues I haven't personally experienced.

JMC47 commented:
Yeah, never trust the wiki.  What it actually means is that D3D is slower on their computer which leads to dualcore causing a crash.

MaJoRoesch commented:
Please don't pull things from the wiki! It's user generated and some bad things can get on there. Me and the other wiki admins do our best to cull silly things, but there's a lot and we can't test to verify everything. :/

MarioManTAW commented:
I will make a mental note to test it out myself, then.

JosJuice commented:
Pulling settings from the wiki without any further testing or knowledge isn't a good way to make game INI PRs at all. Some of the recommended settings on the wiki shouldn't be forced on by game INIs because they have negative effects on other things (e.g. performance), and some of them just don't make sense. I've commented on some of the problems in this PR, but there are probably more.

Also, please make your changes in one commit, not 14.

---

Review status: 0 of 13 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[Data/Sys/GameSettings/R8P.ini, line 23 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3735#-KCtVoKNQ3Y19_pYEBO8:-KCtVoKNQ3Y19_pYEBO9:-849176816) ([raw file](https://github.com/dolphin-emu/dolphin/blob/4861393db7a2d151e1cf9589f416208c5f0a18c7/Data/Sys/GameSettings/R8P.ini#L23)):*
This isn't just some stability fix. This is functionality that's required for the game to function.

---

*[Data/Sys/GameSettings/RSB.ini, line 25 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3735#-KCtVyA_77ITtv3Z62wg:-KCtVyA_77ITtv3Z62wh:1827617107) ([raw file](https://github.com/dolphin-emu/dolphin/blob/4861393db7a2d151e1cf9589f416208c5f0a18c7/Data/Sys/GameSettings/RSB.ini#L25)):*
So you're going to prevent people from using resolutions higher than native just so Classic results won't have a green box? Sounds like a bad idea.

---

*[Data/Sys/GameSettings/RSB.ini, line 29 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3735#-KCtW9fQ7Midb_S_gaBn:-KCtW9fRg8Sk_3alsf9E:-2044669777) ([raw file](https://github.com/dolphin-emu/dolphin/blob/4861393db7a2d151e1cf9589f416208c5f0a18c7/Data/Sys/GameSettings/RSB.ini#L29)):*
We might not want to force this (and SafeTextureColorSamples). It will significantly affect performance for a small benefit, and some users might not have computers that are fast enough to make that trade-off.

---

*[Data/Sys/GameSettings/RZT.ini, line 22 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3735#-KCtWSxzVkhBLeQyMfv6:-KCtWSy-6SzjvlScdKxi:398841848) ([raw file](https://github.com/dolphin-emu/dolphin/blob/4861393db7a2d151e1cf9589f416208c5f0a18c7/Data/Sys/GameSettings/RZT.ini#L22)):*
DisableFog in a default game INI? Is there an issue report for this? We probably shouldn't set this unless we know what's going on.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3735#-:-KCtWlG8U_JOnaNEuqRx:-1916272871)*
<!-- Sent from Reviewable.io -->

MarioManTAW commented:
Okay, let's try to deal with these one at a time.

I can change the comment.
I didn't know it had that side effect. Will change it back.
Again, unaware of side effects.
This is a game I don't have access to to test, but I know Mario Kart Wii had a similar issue. I could change it to the MKW fix and see if it works.

As for commits, I'm new to the GitHub process. I don't know how to do it at once.

degasus commented on Data/Sys/GameSettings/R64.ini:
```diff
@@ -2,6 +2,7 @@
 
 [Core]
 # Values set here will override the main Dolphin settings.
+CPUThread = False # Prevent CPU/GPU desync
```
Is syncGpu already enough? Forcing single core is a bit rough.

degasus commented on Data/Sys/GameSettings/R7P.ini:
```diff
@@ -2,6 +2,7 @@
 
 [Core]
 # Values set here will override the main Dolphin settings.
+SkipIdle = False # Speedup
```
*Very* unlikely. I guess you want to disable the "SyncOnIdleSkippingHack". Idle skipping is a safe enhancement, and I still haven't got why we allow to disable it at all.

degasus commented on Data/Sys/GameSettings/R8P.ini:
```diff
@@ -18,7 +19,5 @@ EmulationIssues = Needs Efb to Ram for BBox (proper graphics).
-[Video_Hacks]
-BBoxEnable = True
+EFBToTextureEnable = False # Fix transitions
+UseBBox = True # Required for proper functionality
```
I like this name, but right now, the ini option is called "BBoxEnable "

degasus commented on Data/Sys/GameSettings/RK5.ini:
```diff
@@ -17,3 +17,5 @@ EmulationStateId = 5
 # Add action replay cheats here.
 
+[Video_Hacks]
+EFBToTextureEnable = True # Fix shadows
```
Oh? If so, this is likely a fixable bug. Is there an issue report for those shadows being broken on EFB2RAM?

degasus commented on Data/Sys/GameSettings/RM3.ini:
```diff
@@ -19,6 +19,7 @@ EmulationIssues =
 
 [Video_Settings]
 SafeTextureCacheColorSamples = 512
+UseRealXFB = False # Prevent black bar
```
Which one do you want to enforce? Virtual XFB? Or just to prohibit real XFB? I doubt real XFB could be broken while virtual XFB works fine....

degasus commented on Data/Sys/GameSettings/RMC.ini:
```diff
@@ -19,6 +19,11 @@ EmulationIssues =
+
+[Video_Settings]
+MSAA = 0 # Avoid overbloom
+UseXFB = False # Prevent blue box
```
Nope, blue box is already fixed by "EFBEmulateFormatChanges = True"

degasus commented on Data/Sys/GameSettings/RMC.ini:
```diff
@@ -19,6 +19,11 @@ EmulationIssues =
 
 [Video_Hacks]
 EFBEmulateFormatChanges = True
+EFBToTextureEnable = False # Correct lens flare
```
Hm, this one will be a major slowdown. Are you sure this should be enforced for everyone? Maybe just a "EmulationIssues" comment.

degasus commented on Data/Sys/GameSettings/RSB.ini:
```diff
@@ -19,3 +19,6 @@ EmulationIssues = Classic mode score report needs real xfb. Nes masterpieces and
 StereoConvergence = 136
+
+[Video_Hacks]
+EFBAccessEnable = False # Fix online FPS
```
This sounds like a horrible hack. Just get a fast enough computer to still support EFB access ;)

degasus commented on Data/Sys/GameSettings/RSP.ini:
```diff
@@ -2,6 +2,7 @@
 
 [Core]
 # Values set here will override the main Dolphin settings.
+CPUThread = False # Display Mii faces correctly
```
Is syncGpu also fine?

degasus commented on Data/Sys/GameSettings/RZT.ini:
```diff
@@ -1,13 +1,25 @@
+EFBAccessEnable = True # Fix aiming
+
+[Video_Settings]
+DisableFog = True # Prevent blue box
```
Hacky again. Are there other ways?

Linktothepast replied:
Disable fog should be removed, emulate format changes prevents the blue box in a non hacky way without the need to disable fog altogether for the game.

degasus commented on Data/Sys/GameSettings/RZT.ini:
```diff
@@ -1,13 +1,25 @@
+DisableFog = True # Prevent blue box
+
+[Video_Enhancements]
+ForceFiltering = True # Fix shading
```
Also a bit hacky. Is this also required at 1x IR?

degasus commented on Data/Sys/GameSettings/SF8.ini:
```diff
@@ -23,3 +23,7 @@ SafeTextureCacheColorSamples = 512
+
+[Video_Hacks]
+EFBToTextureEnable = False # Prevent flickering/invisible Kongs (OpenGL)
+EFBScaledCopy = False # Prevent flickering/invisible Kongs (Direct3D)
```
This works around a dolphin bug which is already fixed in other PRs.

degasus commented on Data/Sys/GameSettings/SMN.ini:
```diff
@@ -24,3 +24,6 @@ SafeTextureCacheColorSamples = 512
 StereoConvergence = 2211
+
+[Video_Enhancements]
+ForceFiltering = True # Fix shading
```
Also hacky, is this required at 1xIR?

MarioManTAW commented:
Most of these are bugs I have not personally experienced, so I suppose in the future I should only fix issues either I personally see or see bug reports for.
Where exactly is this Sync GPU option? On Wii Sports (RSP.ini), I have personally seen the glitchy Mii faces and seen that single core fixes them. If there is a suitable alternative, though, I would be happy to change it.

JosJuice commented:
It's `SyncGPU` under `[Core]`.

MarioManTAW commented:
Is it available in the GUI? That's what I was asking.

JosJuice commented:
No, only in INIs.

MarioManTAW commented:
Tested and SyncGPU does NOT fix the problem. Although functionally the game is perfect (as far as I can tell), the graphical glitches are very distracting.

Linktothepast commented:
Tried to review this and degasus comments are spot on. I would just like to add that in wii sports resort  EFBEmulateFormatChanges = True fixes the blue box already in a non hacky way and you enabled both that and disabled fog together. Disable fog is hacky and moreover having both options is pointless. And i wouldn't enforce force filtering in any title, it is not a compatibility option but an enhancement and can also cause issues or blur the image by default where it is not wanted by the user. Better someone finds the issue behind it or leave it to the user than enable it by default in any title.

MarioManTAW commented:
Most of the issues are now fixed.

What is this SyncOnIdleSkippingHack?
Commented on Wii Sports issue.
Mentioned possible alternative to disabling fog earlier.

MarioManTAW commented:
Anyone looking over this?

Linktothepast commented on Data/Sys/GameSettings/SF8.ini:
```diff
@@ -23,3 +23,5 @@ SafeTextureCacheColorSamples = 512
 [Video_Stereoscopy]
 StereoConvergence = 26
+
+[Video_Hacks]
```
Is there any point to change a file to add a line that does nothing?

Linktothepast commented on Data/Sys/GameSettings/SMN.ini:
```diff
@@ -24,3 +24,5 @@ SafeTextureCacheColorSamples = 512
 #Only affects overworld map
 StereoConvergence = 2211
+
+[Video_Enhancements]
```
Same as above here this line added does nothing.

Linktothepast commented on Data/Sys/GameSettings/RSB.ini:
```diff
@@ -19,3 +19,5 @@ EmulationIssues = Classic mode score report needs real xfb. Nes masterpieces and
 [Video_Stereoscopy]
 StereoConvergence = 136
+
+[Video_Hacks]
```
Added lines do nothing.

MarioManTAW commented:
Any other issues?

Linktothepast commented on Data/Sys/GameSettings/RM3.ini:
```diff
@@ -19,6 +19,7 @@ EmulationIssues =
 
 [Video_Settings]
 SafeTextureCacheColorSamples = 512
+UseRealXFB = True # Prevent black bar
```
You want to enforce real xfb to metroid prime 3 for the black bar issue? Please add:
UseXFB = True
To enforce virtual xfb instead and set the line you added to false.
The way you did it was also false, see my thread of gameini settings in the forums on why real and virtual xfb settings should be used always together in the gameinis, the way you did it here if someone enabled virtual xfb in his general settings it would switch to real xfb otherwise nothing would happen.

JosJuice commented on Data/Sys/GameSettings/R8P.ini:
```diff
@@ -2,6 +2,7 @@
 
 [Core]
 # Values set here will override the main Dolphin settings.
+GFXBackend = OGL # Stability
```
This doesn't seem right. Refer to what JMC posted for details.

Linktothepast commented on Data/Sys/GameSettings/RMK.ini:
```diff
@@ -19,3 +19,6 @@ EmulationIssues =
 EFBEmulateFormatChanges = True
+
+[Video_Settings]
+MSAA = 0 # Fix shadow transparency
```
I don't thing it is a good idea to disable antialiasing because it's results depends on the graphic drivers and the devs do various changes sometimes to antialiasing that might affect it. Would you be here in case something changes and track it to change it back? Also this also disables SSAA, are you sure both msaa and ssaa have an issue?

MarioManTAW commented:
After testing Super Paper Mario (R8P), I see no reason to use OGL over D3D. Change reverted.
I do not have access to Mario Sports Mix (RMK) to know what would be best, but the wiki says Anti-Aliasing: None.

JosJuice commented:
That the wiki says so is not a valid reason on its own. Also, there are still several unresolved comments other than that one.

Linktothepast commented on Data/Sys/GameSettings/RM3.ini:
```diff
@@ -19,7 +19,7 @@ EmulationIssues =
 
 [Video_Settings]
 SafeTextureCacheColorSamples = 512
-UseRealXFB = True # Prevent black bar
```
You should also set userealxfb to false instead of removing it. As it is xfb will be enabled but whether real or virtual is used will depend on what the user have set it last time in the general settings.

Linktothepast commented on Data/Sys/GameSettings/RK5.ini:
```diff
@@ -17,3 +17,4 @@ EmulationStateId = 5
 [ActionReplay]
 # Add action replay cheats here.
 
+[Video_Hacks]
```
This line does nothing and changes a file for no apparent reason.

MarioManTAW commented:
Removed the empty code and added the UseRealXFB = False

MarioManTAW commented:
Are there any other outstanding issues?

JosJuice commented:
Yes, the ones you can see in GitHub's "Files changed" tab.

MarioManTAW commented:
Fixed. On @degasus comment on Wii Sports, I tried it and SyncGPU does **not** fix the problem.

MarioManTAW commented:
Anything else?

JMC47 commented:
Turning off idle-skipping as a speedhack is not acceptable.

degasus commented:
@JMC47 This is not done, only the syncing is disabled. So no impact on single core, but yes, maybe stability issues on dual core.

JMC47 commented:
Okay, yeah, sync on idle skip doesn't matter for single core.

MarioManTAW commented:
Do I need to change something?

lioncash commented:
Consider squashing your changes:

See [here](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/) or [here](https://gist.github.com/lioncash/fbf57f875575d3e13484) on how to do that

Linktothepast commented on Data/Sys/GameSettings/RK5.ini:
```diff
@@ -16,4 +16,3 @@ EmulationStateId = 5
 
 [ActionReplay]
 # Add action replay cheats here.
-
```
You removed a line in this file with no other change please revert.

Linktothepast commented:
Also please add these 2 files : 
https://mega.nz/#!toIyHLYT!FOhvmOmOsx944Rk2VY4OKEVXEoBFj57oKULvYOZoRC8
https://mega.nz/#!s5JwlA7T!T4rgsL5hcbkeEyrKv-OQPemoSe3d9AtrrjYmrkW4P1E

MarioManTAW commented:
Added the files, still trying to figure out how to squash.

JosJuice commented:
I only see one file added.

MarioManTAW commented:
There we go.

JosJuice commented:
The latest commit doesn't add any files, it just contains a one-line change with no visible difference. Is the change that é is encoded differently?

MarioManTAW commented:
I don't actually know what Linktothepast did to the file. I just included it.

MarioManTAW commented:
That was supposed to merge commits.
It failed.
That was a waste of time.
Anyone know how to revert a commit?

MarioManTAW commented:
Never mind, reverted.

JosJuice commented:
Merge commits are not allowed in PRs. You're supposed to rebase instead.

MarioManTAW commented:
I did use rebase. I just didn't use the correct terminology. It took like half an hour and filled the code with junk, so I reverted it.

JosJuice commented:
Can you describe how you rebased so we can figure out went wrong? A correct rebase will be necessary for getting this PR merged.

MarioManTAW commented:
I rebased, then when prompted, I removed all the changes other people added before me, changed all the "pick"s except the first into "squash"s, and saved. I did "git rebase --continue" and it said there was an error. I fixed it and repeated. I repeated these last two steps for 70+ times then it said it was successful.

Linktothepast commented:
The nan.ini file i uploaded removed an emulation issue comment, the file you uploaded doesn't. There is no timer bug anymore in that game.

MarioManTAW commented:
There it goes.

JosJuice commented:
The conflicts in RUU.ini need to be fixed.

MarioManTAW commented:
Apparently, the revert didn't affect that file. Fixed now.

MarioManTAW commented:
And I removed some duplicate code that somehow got in the file.

JMC47 commented:
Looks mostly good to me now. The two things I'm not okay with are the sync on idle skipping thing, and Virtual XFB no longer fixes the black bar in Metroid Prime 2/3/Trilogy.  That's outdated information; when it was made more accurate it started happening again unfortunately.

Wii Music I manually verified to need syncGPU or single core, the other changes are all pretty good based on what I know.  Forcing RealXFB for videos is iffy, sometimes we do it sometimes we don't, but, in this case it probably is fine.

MarioManTAW commented:
So on those issues, what should I do to fix them? Those are games I don't have access to personally.

JMC47 commented:
Just remove those two changes (Virtual XFB for Metroid Prime 3 and the Sync on Idle Skipping for performance in another game) and it lgtm.

MarioManTAW commented:
Fixed.

JosJuice commented:
The commits still need to be squashed.

MarioManTAW commented:
I don't know how to do that, sorry.

JosJuice commented:
I've made the new PR #3768 with the commits of this PR squashed.

MarioManTAW commented:
Okay. Thanks.
